src/util/timer.h

```diff
@@ -33,103 +35,59 @@ class Timer {
   protected:
     QString m_key;
     Stat::ComputeFlags m_compute;
-    bool m_running;
     PerformanceTimer m_time;
 };
 
-class SuspendableTimer : public Timer {
-  public:
-    SuspendableTimer(const QString& key,
-            Stat::ComputeFlags compute = kDefaultComputeFlags);
-    void start();
-    mixxx::Duration suspend();
-    void go();
-    mixxx::Duration elapsed(bool report);
-
-  private:
-    mixxx::Duration m_leapTime;
-};
-
+// TODO: replace with std::experimental::scope_exit<Timer> once stabilized
 class ScopedTimer {
   public:
-    ScopedTimer(const char* key, int i,
-                Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : m_pTimer(NULL),
-              m_cancel(false) {
-        if (CmdlineArgs::Instance().getDeveloper()) {
-            initialize(QString(key), QString::number(i), compute);
-        }
+    ScopedTimer(QStringView key,
+            Stat::ComputeFlags compute = kDefaultComputeFlags)
+            : ScopedTimer(key, QStringView(), compute) {
     }
-
-    ScopedTimer(const char* key, const char *arg = NULL,
-                Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : m_pTimer(NULL),
-              m_cancel(false) {
-        if (CmdlineArgs::Instance().getDeveloper()) {
-            initialize(QString(key), arg ? QString(arg) : QString(), compute);
-        }
+    ScopedTimer(QStringView key,
+            int i,
+            Stat::ComputeFlags compute = kDefaultComputeFlags)
+            : ScopedTimer(key,
+                      CmdlineArgs::Instance().getDeveloper()
+                              ? QString::number(i)
+                              : QStringView(),
+                      compute) {
     }
 
-    ScopedTimer(const char* key, const QString& arg,
-                Stat::ComputeFlags compute = kDefaultComputeFlags)
-            : m_pTimer(NULL),
-              m_cancel(false) {
-        if (CmdlineArgs::Instance().getDeveloper()) {
-            initialize(QString(key), arg, compute);
+    ScopedTimer(QStringView key, QStringView arg, Stat::ComputeFlags compute = kDefaultComputeFlags)
+            : m_maybeTimer(std::nullopt) {
+        if (!CmdlineArgs::Instance().getDeveloper()) {
+            return;
         }
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+        QString strKey = arg.isEmpty() ? key.toString() : key.arg(arg);
+#else
+        QString strKey = arg.isEmpty() ? key.toString() : key.toString().arg(arg);
+#endif
+        m_maybeTimer = std::make_optional<Timer>(std::move(strKey), compute);
+        m_maybeTimer->start();
     }
 
-    virtual ~ScopedTimer() {
-        if (m_pTimer) {
-            if (!m_cancel) {
-                m_pTimer->elapsed(true);
-            }
-            m_pTimer->~Timer();
+    ~ScopedTimer() noexcept {
+        if (m_maybeTimer) {
+            m_maybeTimer->elapsed(true);
         }
+        m_maybeTimer.reset();
```

Comment:
This will happen automatically.

Reply:
Are you sure? Thats what I was asking in this comment: https://github.com/mixxxdj/mixxx/pull/11807#discussion_r1292249067

Reply:
Yes, the back trace with the warning [pr](https://github.com/mixxxdj/mixxx/pull/11807#discussion_r1292351607) proves it. 
But there seems to be a related issue causing the warnig.

Reply:
I cannot confirm that warning with valgrind, It seems to be a false positive.